### PR TITLE
Rnn scratch fix

### PR DESCRIPTION
**Pull request by abhinavsp0730**
*Description of changes:*
pt/mxnet's SparseCategoricalCrossentropy loss is softmax + crossentropy loss. But in Tensorflow by default the SparseCategoricalCrossentropy loss is just crossentropy loss. So, in order to make it softmax + crossentropy we've to pass ```from_logits=True``` in the function's argument.

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-en/PR-1450/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1450/

**Comment by mli** [comment]
Job d2l-en/PR-1450/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1450/

**Review by terrytangyuan — APPROVED** [review]
